clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -440,16 +440,36 @@ private boolean renameObject(LakeFSFileStatus srcStatus, Path dst) throws IOExce
 
         ObjectsApi objects = lfsClient.getObjects();
         //TODO (Tals): Can we add metadata? we currently don't have an API to get the metadata of an object.
-        ObjectStageCreation creationReq = new ObjectStageCreation()
-                .checksum(srcStatus.getChecksum())
-                .sizeBytes(srcStatus.getLen())
-                .physicalAddress(srcStatus.getPhysicalAddress());
+        ObjectCopyCreation creationReq = new ObjectCopyCreation()
+                .srcRef(srcObjectLoc.getRef())
+                .srcPath(srcObjectLoc.getPath());
 
+        Boolean fallbackToStage = false;
         try {
-            objects.stageObject(dstObjectLoc.getRepository(), dstObjectLoc.getRef(), dstObjectLoc.getPath(),
+            objects.copyObject(dstObjectLoc.getRepository(), dstObjectLoc.getRef(), dstObjectLoc.getPath(),
                     creationReq);
         } catch (ApiException e) {
-            throw translateException("renameObject: src:" + srcStatus.getPath() + ", dst: " + dst + ", failed to stage object", e);
+            if (e.getCode() !=HttpStatus.SC_INTERNAL_SERVER_ERROR ||
+                    e.getMessage().contains("invalid API endpoint")){
+                throw translateException("renameObject: src:" + srcStatus.getPath() + ", dst: " + dst + ", failed to copy object", e);
+            }
+
+            LOG.warn("Copy API doesn't exist, falling back to stageObject");
+            fallbackToStage = true;
```

Comment:
This **must** be stored for future calls.  Otherwise whenever we run a new client with an old version, we will double the number of API calls.  That gives poor performance, spams logs on client and on server, and eats away from any API call quota.